activation/validation.go

```diff
@@ -218,21 +210,12 @@ func (*Validator) NIPostChallenge(challenge *types.NIPostChallenge, atxs atxProv
 	if prevATX.Sequence+1 != challenge.Sequence {
 		return fmt.Errorf("sequence number is not one more than prev sequence number")
 	}
-
-	if challenge.InitialPost != nil {
-		return fmt.Errorf("prevATX declared, but initial Post is included in challenge")
-	}
-
-	if challenge.CommitmentATX != nil {
-		return fmt.Errorf("prevATX declared, but commitmentATX is included")
-	}
-
 	return nil
 }
 
-func (*Validator) PositioningAtx(id *types.ATXID, atxs atxProvider, goldenATXID types.ATXID, pubepoch types.EpochID, layersPerEpoch uint32) error {
+func (v *Validator) PositioningAtx(id *types.ATXID, atxs atxProvider, goldenATXID types.ATXID, pubepoch types.EpochID) error {
 	if *id == types.EmptyATXID {
-		return fmt.Errorf("empty positioning atx")
+		v.log.Panic("empty positioning atx")
```

Comment:
ditto.

Btw, if `id` cannot be `nil`, then perhaps it should be passed by value to avoid the possibility of dereferencing a `nil` at all? The compiler should be able to optimize the copy away anyway.

Reply:
changed. thanks